shell/common/shell.cc

```diff
@@ -1404,6 +1404,17 @@ bool Shell::OnServiceProtocolRunInView(
   configuration.AddAssetResolver(
       std::make_unique<DirectoryAssetBundle>(fml::OpenDirectory(
           asset_directory_path.c_str(), false, fml::FilePermission::kRead)));
+  // Add the original asset directory to the resolvers so that unmodified assets
+  // bundled with the application specific format (APK, IPA) can be used without
+  // syncing to the Dart devFS.
+  if (fml::UniqueFD::traits_type::IsValid(settings_.assets_dir)) {
+    configuration.AddAssetResolver(std::make_unique<DirectoryAssetBundle>(
```

Comment:
Can the code that constructs this asset resolver be broken out into a function shared between this and OnServiceProtocolSetAssetBundlePath?

Reply:
I added a function that creates the resolver, PTAL and let me know if I'm following the proper flutter engine/c++ idioms